### ISSUE TITLE
SynthEBD crashes after running patch for second time

### ISSUE THREAD

**Issue by leonardovcr** [Synthesis-Collective/SynthEBD#47]
Hi, another crash is happening if the patcher is run a second time in a row. Removing the Consistency file and running the patcher again solves the problem.

SynthEBD has crashed with the following error:
An object implementing IHandleObservableErrors (often a ReactiveCommand or ObservableAsPropertyHelper) has errored, thereby breaking its observable pipeline. To prevent this, ensure the pipeline does not error, or Subscribe to the ThrownExceptions property of the object in question to handle the erroneous case.
   at ReactiveUI.RxApp.<>c__DisplayClass9_0.<.cctor>b__2() in /_/src/ReactiveUI/RxApp.cs:line 109
   at System.Reactive.Concurrency.Scheduler.Invoke(Action action) in /_/Rx.NET/Source/src/System.Reactive/Concurrency/Scheduler.Simple.cs:line 253
   at System.Reactive.Concurrency.Scheduler.<>c.<Schedule>b__74_0(IScheduler _, Action a) in /_/Rx.NET/Source/src/System.Reactive/Concurrency/Scheduler.Simple.cs:line 37
   at System.Reactive.Concurrency.DispatcherScheduler.<>c__DisplayClass12_0`1.<Schedule>b__0() in /_/src/ReactiveUI.Uwp/Rx/Concurrency/DispatcherScheduler.cs:line 96
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

An item with the same key has already been added. Key: 01C9F7:Skyrim.esm
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at SynthEBD.Patcher.MainLoop(IEnumerable`1 npcCollection, Boolean skipLinkedSecondaryNPCs, SkyrimMod outputMod, CategorizedFlattenedAssetPacks sortedAssetPacks, BodyGenConfigs bodyGenConfigs, Settings_OBody oBodySettings, HeightConfig currentHeightConfig, Settings_Headparts headPartSettings, Int32 npcCounter, HashSet`1 generatedLinkGroups, HashSet`1 skippedLinkedNPCs, Keyword EBDFaceKW, Keyword EBDScriptKW, Spell bodySlideAssignmentSpell, HashSet`1 headPartNPCs) in D:\a\SynthEBD\SynthEBD\SynthEBD\Patcher\Patcher.cs:line 555
   at SynthEBD.Patcher.RunPatcher() in D:\a\SynthEBD\SynthEBD\SynthEBD\Patcher\Patcher.cs:line 180
   at SynthEBD.VM_RunButton.<>c__DisplayClass2_0.<<-ctor>b__0>d.MoveNext() in D:\a\SynthEBD\SynthEBD\SynthEBD\RunButton\VM_RunButton.cs:line 49

**Comment by Piranha91** [comment]
Thanks, I think this might be the same issue as was reported on the Discord. I’ll take a look. In the meantime, could you please tell me your steps to get around the “secure folders” issue you had? I think others may be having the same problem.

**Comment by Piranha91** [comment]
Ok, have time to take a closer look now - this is a different issue. Can you clarify, does it happen every 2nd time you run the patcher, or did it happen only once when upgrading to a new version of SynthEBD?

**Comment by leonardovcr** [comment]
About the secure folders, its Windows AntiVirus causing problems. I had to add SynthEBD folders and process to the Windows Defender Exclusion List, like this:

![image](https://user-images.githubusercontent.com/13605450/187052869-3135afbe-462a-4bf5-9596-b3acc91c5bcb.png)

Now for the crash, I can reproduce it by running the patcher again immediatly after it finishes patching and writing its output. If I run the patch, close the app, open it again and run the patch again, the problem doesn't happen. I think it started to happen after I added a new "Specific NPC assignment" with a forced asset pack...
